flink-java/src/main/java/org/apache/flink/api/java/DataSet.java

```diff
@@ -1343,10 +1343,34 @@ public long count() throws Exception {
 	// --------------------------------------------------------------------------------------------
 	
 	/**
-	 * Writes a DataSet as a text file to the specified location.<br>
-	 * For each element of the DataSet the result of {@link Object#toString()} is written.  
+	 * Writes a DataSet as text file(s) to the specified location.<br>
+	 * For each element of the DataSet the result of {@link Object#toString()} is written.<br/>
+	 * <br/>
+	 * <span class="strong">How it writes DataSet</span><br/>
+	 * A single file is written when
+	 * <ul>
+	 *   <li>parallelism is set to 1</li>
+	 *   <li>or <a href="https://ci.apache.org/projects/flink/flink-docs-master/setup/config.html#file-systems">fs.output.always-create-directory</a> is set to true in flink-conf.yaml file</li>
```

Comment:
@StephanEwen , good catch. 
Let me clarify this further. We have 3 cases (instead of the current 2 cases)
1. /path1 as text file
2. /path1/1 and /path1/2 and so on 
3. /path1/1 only. and 1 is a sole text file.
I mixed 1 and 3. will fix